django/db/models/query.py

```diff
@@ -322,6 +322,9 @@ def aggregate(self, *args, **kwargs):
         """
         if self.query.distinct_fields:
             raise NotImplementedError("aggregate() + distinct(fields) not implemented.")
+        if (not all(isinstance(kwarg, (BaseExpression, Combinable)) for kwarg in kwargs.values()) or
+                not all(isinstance(arg, (BaseExpression, Combinable)) for arg in args)):
```

Comment:
In the past, sometimes ducktyping has proved more useful than isinstance. Not sure if something like checking for resolve_expression/default_alias attributes might be preferred. Any thoughts @jarshwah?